src/onnx/parse_softmaxcrossentropyloss.cpp

```
@@ -0,0 +1,420 @@
+
+output (differentiable) : T
+    Weighted loss float Tensor. If reduction is 'none', this has the shape of [batch_size],
+    or [batch_size, D1, D2, ..., Dk] in case of K-dimensional loss. Otherwise, it is a scalar.
+
+log_prob (optional, differentiable) : T
+    Log probability tensor. If the output of softmax is prob, its value is log(prob).
+
+Type Constraints
+===================
+    T : tensor(float16), tensor(float), tensor(double), tensor(bfloat16) (Currently not supported in
+MIGX) Constrain input and output types to float tensors.
+
+Tind : tensor(int32), tensor(int64)
+    Constrain target to integer types
+*/
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
+{
+    std::vector<op_desc> operators() const { return {{"SoftmaxCrossEntropyLoss"}}; }
+
+    // Handle ignore index if it's within range of allowable classes
+    // return false if ignore index out of bounds and never add literal to graph
+    // return true if ignore index is in bound and pass back literal
+    bool normalize_input_index(const onnx_parser& parser,
+                               const onnx_parser::node_info& info,
+                               const int64_t input_classes,
+                               instruction_ref& ignore_index) const
+    {
+        bool has_ignore_index = contains(info.attributes, "ignore_index");
+        if(has_ignore_index)
+        {
+            auto ignore_index_val =
+                parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
+
+            if(ignore_index_val >= input_classes || abs(ignore_index_val) > input_classes)
+            {
+                return false;
+            }
+
+            if(ignore_index_val < 0)
```

This shouldn't be done. It should be considered as out of bounds, when the index value is not in the range `[0, C)`.

---

The odd part about this is that when I test for gold values in numpy and Onnxruntime they both give me consistent and valid outputs when I read in the model with the same inputs/in bound ignore index if I take the absolute value

---

Nevermind. went over the test. You are correct. Ive changed this to be consistent. Looks like there was a  bad value in the test ( zero) where there should be an entry. numpy and onnxruntime won't error out ,but infact just flat out silently ignore the negative input and operate as if ignore index doesn't exist.